src/lib/y2partitioner/widgets/encrypt_method_options.rb

```diff
@@ -249,11 +287,95 @@ def exist_secure_key?
         !controller.secure_key.nil?
       end
 
-      # Whether there are several available APQNs
+      # Whether the AES master key can be chosen
+      #
+      # The master key can be chosen when there are several keys available and the device does not
+      # have an associated secure key yet.
+      #
+      # @return [Boolean]
+      def select_master_key?
+        apqns_by_key.keys.size > 1
+      end
+
+      # Whether there is any possibility to define the APQNs
       #
       # @return [Boolean]
-      def several_apqns?
-        controller.online_apqns.size > 1
+      def select_apqns?
+        apqns_by_key.any? { |i| i.last.size > 1 }
+      end
+
+      # @return [String]
+      def initial_key
+        @initial_key ||=
+          if controller.apqns.empty?
+            pervasive_keys.min_by { |k| apqns_by_key[k].first.name }
+          else
+            find_key_for(controller.apqns.first)
+          end
+      end
+
+      # Master key configured at the given APQN
+      #
+      # @param apqn [Y2Storage::EncryptionProcesses::Apqn]
+      # @return [String]
```

Comment:
can it return nil?

Reply:
No, it cannot. Only APQNs with a master key are considered by the Partitioner.